### ISSUE TITLE
SLPVectorizer: Assertion `isVectorLikeInstWithConstOps(FirstInst) && isVectorLikeInstWithConstOps(I) && "Expected vector-like insts only."' failed with -O3 -march=skx

### ISSUE THREAD

**Issue by Vsevolod-Livinskij** [llvm/llvm-project#58616]
I've attached C++ and LLVM IR reproducers.
clang version 16.0.0 (https://github.com/llvm/llvm-project.git b50392342be27efdc01d877ba0c357485dc7e918)

CC: @regehr @dbabokin

IR reproducer:
```llvm
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-unknown-linux-gnu"

; Function Attrs: mustprogress
define void @_Z1nbPlPA5_iS_PA2_A5_A12_j(i1 noundef %d, ptr %f, ptr %o) #0 {
entry:
  br label %for.cond

for.cond:                                         ; preds = %for.inc65, %entry
  %j.0 = phi i32 [ 0, %entry ], [ %add66, %for.inc65 ]
  %cmp = icmp ult i32 %j.0, 13
  br i1 %cmp, label %for.body, label %for.end67

for.body:                                         ; preds = %for.cond
  %idxprom = zext i32 %j.0 to i64
  %cond.in.v = select i1 %d, ptr null, ptr %f
  %cond.in = getelementptr i64, ptr %cond.in.v, i64 %idxprom
  %cond = load i64, ptr %cond.in, align 8
  %tobool3.not = icmp eq i64 %cond, 0
  br i1 %tobool3.not, label %cond.false5, label %cond.end8

cond.false5:                                      ; preds = %for.body
  %0 = load i8, ptr %o, align 1
  %conv = sext i8 %0 to i64
  br label %cond.end8

cond.end8:                                        ; preds = %cond.false5, %for.body
  %cond9 = phi i64 [ %conv, %cond.false5 ], [ 0, %for.body ]
  %tobool10.not = icmp eq i64 %cond9, 0
  %conv11 = zext i1 %tobool10.not to i32
  br label %for.cond12

for.cond12:                                       ; preds = %for.inc62, %cond.end8
  %k.0 = phi i32 [ %conv11, %cond.end8 ], [ %add63, %for.inc62 ]
  %cmp13 = icmp ult i32 %k.0, 2
  br i1 %cmp13, label %for.body15, label %for.inc65

for.body15:                                       ; preds = %for.cond12
  %1 = load i32, ptr %f, align 4
  br label %for.cond21

for.cond21:                                       ; preds = %for.cond25, %for.body15
  %2 = and i32 %1, 1
  %tobool22.not = icmp eq i32 %2, 0
  br i1 %tobool22.not, label %for.inc62, label %for.cond25

for.cond25:                                       ; preds = %for.body29, %for.cond21
  br i1 %d, label %for.body29, label %for.cond21

for.body29:                                       ; preds = %for.cond25
  store i8 0, ptr %f, align 1
  br label %for.cond25

for.inc62:                                        ; preds = %for.cond21
  %add63 = add i32 %k.0, 1
  br label %for.cond12

for.inc65:                                        ; preds = %for.cond12
  %add66 = add i32 %j.0, 4
  br label %for.cond

for.end67:                                        ; preds = %for.cond
  ret void
}

attributes #0 = { mustprogress }
```

Error:
`>$ clang++ -c -march=skx -O3 reduced.ll`
```console
clang++: /testing/llvm/llvm_src_main/llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp:7736: llvm::slpvectorizer::BoUpSLP::getLastInstructionInBundle(const llvm::slpvectorizer::BoUpSLP::TreeEntry*)::<lambda()>: Assertion `isVectorLikeInstWithConstOps(FirstInst) && isVectorLikeInstWithConstOps(I) && "Expected vector-like insts only."' failed.
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.	Program arguments: clang++ -c -march=skx -O3 reduced.ll
1.	Optimizer
 #0 0x00005562ef6f1c95 PrintStackTraceSignalHandler(void*) Signals.cpp:0:0
 #1 0x00005562ef6ef8c4 llvm::sys::CleanupOnSignal(unsigned long) (/testing/llvm/bin_main/bin/clang-16+0x23c18c4)
 #2 0x00005562ef6390a8 CrashRecoverySignalHandler(int) CrashRecoveryContext.cpp:0:0
 #3 0x00007f3f40944520 (/lib/x86_64-linux-gnu/libc.so.6+0x42520)
 #4 0x00007f3f40998a7c pthread_kill (/lib/x86_64-linux-gnu/libc.so.6+0x96a7c)
 #5 0x00007f3f40944476 gsignal (/lib/x86_64-linux-gnu/libc.so.6+0x42476)
 #6 0x00007f3f4092a7f3 abort (/lib/x86_64-linux-gnu/libc.so.6+0x287f3)
 #7 0x00007f3f4092a71b (/lib/x86_64-linux-gnu/libc.so.6+0x2871b)
 #8 0x00007f3f4093be96 (/lib/x86_64-linux-gnu/libc.so.6+0x39e96)
 #9 0x00005562ef9af0bf llvm::slpvectorizer::BoUpSLP::getLastInstructionInBundle(llvm::slpvectorizer::BoUpSLP::TreeEntry const*) (/testing/llvm/bin_main/bin/clang-16+0x26810bf)
#10 0x00005562ef9af1f2 llvm::slpvectorizer::BoUpSLP::setInsertPointAfterBundle(llvm::slpvectorizer::BoUpSLP::TreeEntry const*) (/testing/llvm/bin_main/bin/clang-16+0x26811f2)
#11 0x00005562ef9b58f9 llvm::slpvectorizer::BoUpSLP::vectorizeTree(llvm::slpvectorizer::BoUpSLP::TreeEntry*) (/testing/llvm/bin_main/bin/clang-16+0x26878f9)
#12 0x00005562ef9b8a4b llvm::slpvectorizer::BoUpSLP::vectorizeTree(llvm::ArrayRef<llvm::Value*>) (/testing/llvm/bin_main/bin/clang-16+0x268aa4b)
#13 0x00005562ef9b6574 llvm::slpvectorizer::BoUpSLP::vectorizeTree(llvm::slpvectorizer::BoUpSLP::TreeEntry*) (/testing/llvm/bin_main/bin/clang-16+0x2688574)
#14 0x00005562ef9b8a4b llvm::slpvectorizer::BoUpSLP::vectorizeTree(llvm::ArrayRef<llvm::Value*>) (/testing/llvm/bin_main/bin/clang-16+0x268aa4b)
#15 0x00005562ef9b5733 llvm::slpvectorizer::BoUpSLP::vectorizeTree(llvm::slpvectorizer::BoUpSLP::TreeEntry*) (/testing/llvm/bin_main/bin/clang-16+0x2687733)
#16 0x00005562ef9ca9a9 llvm::slpvectorizer::BoUpSLP::vectorizeTree(llvm::MapVector<llvm::Value*, llvm::SmallVector<llvm::Instruction*, 2u>, llvm::DenseMap<llvm::Value*, unsigned int, llvm::DenseMapInfo<llvm::Value*, void>, llvm::detail::DenseMapPair<llvm::Value*, unsigned int>>, std::vector<std::pair<llvm::Value*, llvm::SmallVector<llvm::Instruction*, 2u>>, std::allocator<std::pair<llvm::Value*, llvm::SmallVector<llvm::Instruction*, 2u>>>>>&) (/testing/llvm/bin_main/bin/clang-16+0x269c9a9)
#17 0x00005562ef9cd2c8 llvm::slpvectorizer::BoUpSLP::vectorizeTree() (/testing/llvm/bin_main/bin/clang-16+0x269f2c8)
#18 0x00005562ef9d8400 llvm::SLPVectorizerPass::tryToVectorizeList(llvm::ArrayRef<llvm::Value*>, llvm::slpvectorizer::BoUpSLP&, bool) (/testing/llvm/bin_main/bin/clang-16+0x26aa400)
#19 0x00005562ef9db395 llvm::SLPVectorizerPass::vectorizeSimpleInstructions(llvm::SmallSetVector<llvm::Instruction*, 8u>&, llvm::BasicBlock*, llvm::slpvectorizer::BoUpSLP&, bool) (/testing/llvm/bin_main/bin/clang-16+0x26ad395)
#20 0x00005562ef9de33e llvm::SLPVectorizerPass::vectorizeChainsInBlock(llvm::BasicBlock*, llvm::slpvectorizer::BoUpSLP&) (/testing/llvm/bin_main/bin/clang-16+0x26b033e)
#21 0x00005562ef9e1b50 llvm::SLPVectorizerPass::runImpl(llvm::Function&, llvm::ScalarEvolution*, llvm::TargetTransformInfo*, llvm::TargetLibraryInfo*, llvm::AAResults*, llvm::LoopInfo*, llvm::DominatorTree*, llvm::AssumptionCache*, llvm::DemandedBits*, llvm::OptimizationRemarkEmitter*) (.part.0) SLPVectorizer.cpp:0:0
#22 0x00005562ef9e2950 llvm::SLPVectorizerPass::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) (/testing/llvm/bin_main/bin/clang-16+0x26b4950)
#23 0x00005562f0d80d76 llvm::detail::PassModel<llvm::Function, llvm::SLPVectorizerPass, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Function>>::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) (/testing/llvm/bin_main/bin/clang-16+0x3a52d76)
#24 0x00005562efaa46cc llvm::detail::PassModel<llvm::Function, llvm::PassManager<llvm::Function, llvm::AnalysisManager<llvm::Function>>, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Function>>::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) (/testing/llvm/bin_main/bin/clang-16+0x27766cc)
#25 0x00005562eef33b55 llvm::ModuleToFunctionPassAdaptor::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/testing/llvm/bin_main/bin/clang-16+0x1c05b55)
#26 0x00005562efa92566 llvm::detail::PassModel<llvm::Module, llvm::ModuleToFunctionPassAdaptor, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Module>>::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/testing/llvm/bin_main/bin/clang-16+0x2764566)
#27 0x00005562eef31d2e llvm::PassManager<llvm::Module, llvm::AnalysisManager<llvm::Module>>::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/testing/llvm/bin_main/bin/clang-16+0x1c03d2e)
#28 0x00005562efaa6828 (anonymous namespace)::EmitAssemblyHelper::RunOptimizationPipeline(clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream>>&, std::unique_ptr<llvm::ToolOutputFile, std::default_delete<llvm::ToolOutputFile>>&) BackendUtil.cpp:0:0
#29 0x00005562efaa8ea7 clang::EmitBackendOutput(clang::DiagnosticsEngine&, clang::HeaderSearchOptions const&, clang::CodeGenOptions const&, clang::TargetOptions const&, clang::LangOptions const&, llvm::StringRef, llvm::Module*, clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream>>) (/testing/llvm/bin_main/bin/clang-16+0x277aea7)
#30 0x00005562f097cf8a clang::CodeGenAction::ExecuteAction() (/testing/llvm/bin_main/bin/clang-16+0x364ef8a)
#31 0x00005562f026c0a9 clang::FrontendAction::Execute() (/testing/llvm/bin_main/bin/clang-16+0x2f3e0a9)
#32 0x00005562f01f341e clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/testing/llvm/bin_main/bin/clang-16+0x2ec541e)
#33 0x00005562f0333d38 clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/testing/llvm/bin_main/bin/clang-16+0x3005d38)
#34 0x00005562ee10b3fd cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/testing/llvm/bin_main/bin/clang-16+0xddd3fd)
#35 0x00005562ee105278 ExecuteCC1Tool(llvm::SmallVectorImpl<char const*>&) driver.cpp:0:0
#36 0x00005562f005b229 void llvm::function_ref<void ()>::callback_fn<clang::driver::CC1Command::Execute(llvm::ArrayRef<llvm::Optional<llvm::StringRef>>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>*, bool*) const::'lambda'()>(long) Job.cpp:0:0
#37 0x00005562ef639771 llvm::CrashRecoveryContext::RunSafely(llvm::function_ref<void ()>) (/testing/llvm/bin_main/bin/clang-16+0x230b771)
#38 0x00005562f005bcce clang::driver::CC1Command::Execute(llvm::ArrayRef<llvm::Optional<llvm::StringRef>>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>*, bool*) const (.part.0) Job.cpp:0:0
#39 0x00005562f002552e clang::driver::Compilation::ExecuteCommand(clang::driver::Command const&, clang::driver::Command const*&, bool) const (/testing/llvm/bin_main/bin/clang-16+0x2cf752e)
#40 0x00005562f00260ad clang::driver::Compilation::ExecuteJobs(clang::driver::JobList const&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*>>&, bool) const (/testing/llvm/bin_main/bin/clang-16+0x2cf80ad)
#41 0x00005562f00353cc clang::driver::Driver::ExecuteCompilation(clang::driver::Compilation&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*>>&) (/testing/llvm/bin_main/bin/clang-16+0x2d073cc)
#42 0x00005562ee109be5 clang_main(int, char**) (/testing/llvm/bin_main/bin/clang-16+0xddbbe5)
#43 0x00007f3f4092bd90 (/lib/x86_64-linux-gnu/libc.so.6+0x29d90)
#44 0x00007f3f4092be40 __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x29e40)
#45 0x00005562ee104465 _start (/testing/llvm/bin_main/bin/clang-16+0xdd6465)
clang-16: error: clang frontend command failed with exit code 134 (use -v to see invocation)
clang version 16.0.0 (https://github.com/llvm/llvm-project.git b50392342be27efdc01d877ba0c357485dc7e918)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /testing/llvm/bin_main/bin
```

C++ reproducer:
```c++
extern char a[];
const int &c(int &d, const int &f) { return d ? f : d; }
int e, g;
long h;
char *i;
void n(bool d, long f[], int o[][5], long q[], unsigned s[][2][5][12]) {
  for (int j = 0; j < 13; j += 4)
    for (unsigned k(!((d ? q[j] : f[j]) ?: i[j])); k < 2; k += g) {
      const int &r = c(o[j][g], 0);
      int b = r;
      for (short l(b); l; l += 2)
        for (short m = 1; m < 9; m += 3)
          for (bool p(s[j][g][2][3]); s[j][g][2][3] < (bool)h;)
            a[p] = e;
    }
}
```

Error:
`>$ clang++ -march=skx -O3 -c func.cpp`
```console
clang++: /testing/llvm/llvm_src_main/llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp:7736: llvm::slpvectorizer::BoUpSLP::getLastInstructionInBundle(const llvm::slpvectorizer::BoUpSLP::TreeEntry*)::<lambda()>: Assertion `isVectorLikeInstWithConstOps(FirstInst) && isVectorLikeInstWithConstOps(I) && "Expected vector-like insts only."' failed.
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.	Program arguments: clang++ -march=skx -O3 -c func.cpp
1.	<eof> parser at end of file
2.	Optimizer
 #0 0x000056485f8f7c95 PrintStackTraceSignalHandler(void*) Signals.cpp:0:0
 #1 0x000056485f8f58c4 llvm::sys::CleanupOnSignal(unsigned long) (/testing/llvm/bin_main/bin/clang-16+0x23c18c4)
 #2 0x000056485f83f0a8 CrashRecoverySignalHandler(int) CrashRecoveryContext.cpp:0:0
 #3 0x00007fc0be04a520 (/lib/x86_64-linux-gnu/libc.so.6+0x42520)
 #4 0x00007fc0be09ea7c pthread_kill (/lib/x86_64-linux-gnu/libc.so.6+0x96a7c)
 #5 0x00007fc0be04a476 gsignal (/lib/x86_64-linux-gnu/libc.so.6+0x42476)
 #6 0x00007fc0be0307f3 abort (/lib/x86_64-linux-gnu/libc.so.6+0x287f3)
 #7 0x00007fc0be03071b (/lib/x86_64-linux-gnu/libc.so.6+0x2871b)
 #8 0x00007fc0be041e96 (/lib/x86_64-linux-gnu/libc.so.6+0x39e96)
 #9 0x000056485fbb50bf llvm::slpvectorizer::BoUpSLP::getLastInstructionInBundle(llvm::slpvectorizer::BoUpSLP::TreeEntry const*) (/testing/llvm/bin_main/bin/clang-16+0x26810bf)
#10 0x000056485fbb51f2 llvm::slpvectorizer::BoUpSLP::setInsertPointAfterBundle(llvm::slpvectorizer::BoUpSLP::TreeEntry const*) (/testing/llvm/bin_main/bin/clang-16+0x26811f2)
#11 0x000056485fbbb8f9 llvm::slpvectorizer::BoUpSLP::vectorizeTree(llvm::slpvectorizer::BoUpSLP::TreeEntry*) (/testing/llvm/bin_main/bin/clang-16+0x26878f9)
#12 0x000056485fbbea4b llvm::slpvectorizer::BoUpSLP::vectorizeTree(llvm::ArrayRef<llvm::Value*>) (/testing/llvm/bin_main/bin/clang-16+0x268aa4b)
#13 0x000056485fbbc574 llvm::slpvectorizer::BoUpSLP::vectorizeTree(llvm::slpvectorizer::BoUpSLP::TreeEntry*) (/testing/llvm/bin_main/bin/clang-16+0x2688574)
#14 0x000056485fbbea4b llvm::slpvectorizer::BoUpSLP::vectorizeTree(llvm::ArrayRef<llvm::Value*>) (/testing/llvm/bin_main/bin/clang-16+0x268aa4b)
#15 0x000056485fbbb733 llvm::slpvectorizer::BoUpSLP::vectorizeTree(llvm::slpvectorizer::BoUpSLP::TreeEntry*) (/testing/llvm/bin_main/bin/clang-16+0x2687733)
#16 0x000056485fbd09a9 llvm::slpvectorizer::BoUpSLP::vectorizeTree(llvm::MapVector<llvm::Value*, llvm::SmallVector<llvm::Instruction*, 2u>, llvm::DenseMap<llvm::Value*, unsigned int, llvm::DenseMapInfo<llvm::Value*, void>, llvm::detail::DenseMapPair<llvm::Value*, unsigned int>>, std::vector<std::pair<llvm::Value*, llvm::SmallVector<llvm::Instruction*, 2u>>, std::allocator<std::pair<llvm::Value*, llvm::SmallVector<llvm::Instruction*, 2u>>>>>&) (/testing/llvm/bin_main/bin/clang-16+0x269c9a9)
#17 0x000056485fbd32c8 llvm::slpvectorizer::BoUpSLP::vectorizeTree() (/testing/llvm/bin_main/bin/clang-16+0x269f2c8)
#18 0x000056485fbde400 llvm::SLPVectorizerPass::tryToVectorizeList(llvm::ArrayRef<llvm::Value*>, llvm::slpvectorizer::BoUpSLP&, bool) (/testing/llvm/bin_main/bin/clang-16+0x26aa400)
#19 0x000056485fbe1395 llvm::SLPVectorizerPass::vectorizeSimpleInstructions(llvm::SmallSetVector<llvm::Instruction*, 8u>&, llvm::BasicBlock*, llvm::slpvectorizer::BoUpSLP&, bool) (/testing/llvm/bin_main/bin/clang-16+0x26ad395)
#20 0x000056485fbe433e llvm::SLPVectorizerPass::vectorizeChainsInBlock(llvm::BasicBlock*, llvm::slpvectorizer::BoUpSLP&) (/testing/llvm/bin_main/bin/clang-16+0x26b033e)
#21 0x000056485fbe7b50 llvm::SLPVectorizerPass::runImpl(llvm::Function&, llvm::ScalarEvolution*, llvm::TargetTransformInfo*, llvm::TargetLibraryInfo*, llvm::AAResults*, llvm::LoopInfo*, llvm::DominatorTree*, llvm::AssumptionCache*, llvm::DemandedBits*, llvm::OptimizationRemarkEmitter*) (.part.0) SLPVectorizer.cpp:0:0
#22 0x000056485fbe8950 llvm::SLPVectorizerPass::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) (/testing/llvm/bin_main/bin/clang-16+0x26b4950)
#23 0x0000564860f86d76 llvm::detail::PassModel<llvm::Function, llvm::SLPVectorizerPass, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Function>>::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) (/testing/llvm/bin_main/bin/clang-16+0x3a52d76)
#24 0x000056485fcaa6cc llvm::detail::PassModel<llvm::Function, llvm::PassManager<llvm::Function, llvm::AnalysisManager<llvm::Function>>, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Function>>::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) (/testing/llvm/bin_main/bin/clang-16+0x27766cc)
#25 0x000056485f139b55 llvm::ModuleToFunctionPassAdaptor::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/testing/llvm/bin_main/bin/clang-16+0x1c05b55)
#26 0x000056485fc98566 llvm::detail::PassModel<llvm::Module, llvm::ModuleToFunctionPassAdaptor, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Module>>::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/testing/llvm/bin_main/bin/clang-16+0x2764566)
#27 0x000056485f137d2e llvm::PassManager<llvm::Module, llvm::AnalysisManager<llvm::Module>>::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/testing/llvm/bin_main/bin/clang-16+0x1c03d2e)
#28 0x000056485fcac828 (anonymous namespace)::EmitAssemblyHelper::RunOptimizationPipeline(clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream>>&, std::unique_ptr<llvm::ToolOutputFile, std::default_delete<llvm::ToolOutputFile>>&) BackendUtil.cpp:0:0
#29 0x000056485fcaeea7 clang::EmitBackendOutput(clang::DiagnosticsEngine&, clang::HeaderSearchOptions const&, clang::CodeGenOptions const&, clang::TargetOptions const&, clang::LangOptions const&, llvm::StringRef, llvm::Module*, clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream>>) (/testing/llvm/bin_main/bin/clang-16+0x277aea7)
#30 0x0000564860b820c2 clang::BackendConsumer::HandleTranslationUnit(clang::ASTContext&) (/testing/llvm/bin_main/bin/clang-16+0x364e0c2)
#31 0x0000564861c125a9 clang::ParseAST(clang::Sema&, bool, bool) (/testing/llvm/bin_main/bin/clang-16+0x46de5a9)
#32 0x00005648604720a9 clang::FrontendAction::Execute() (/testing/llvm/bin_main/bin/clang-16+0x2f3e0a9)
#33 0x00005648603f941e clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/testing/llvm/bin_main/bin/clang-16+0x2ec541e)
#34 0x0000564860539d38 clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/testing/llvm/bin_main/bin/clang-16+0x3005d38)
#35 0x000056485e3113fd cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/testing/llvm/bin_main/bin/clang-16+0xddd3fd)
#36 0x000056485e30b278 ExecuteCC1Tool(llvm::SmallVectorImpl<char const*>&) driver.cpp:0:0
#37 0x0000564860261229 void llvm::function_ref<void ()>::callback_fn<clang::driver::CC1Command::Execute(llvm::ArrayRef<llvm::Optional<llvm::StringRef>>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>*, bool*) const::'lambda'()>(long) Job.cpp:0:0
#38 0x000056485f83f771 llvm::CrashRecoveryContext::RunSafely(llvm::function_ref<void ()>) (/testing/llvm/bin_main/bin/clang-16+0x230b771)
#39 0x0000564860261cce clang::driver::CC1Command::Execute(llvm::ArrayRef<llvm::Optional<llvm::StringRef>>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>*, bool*) const (.part.0) Job.cpp:0:0
#40 0x000056486022b52e clang::driver::Compilation::ExecuteCommand(clang::driver::Command const&, clang::driver::Command const*&, bool) const (/testing/llvm/bin_main/bin/clang-16+0x2cf752e)
#41 0x000056486022c0ad clang::driver::Compilation::ExecuteJobs(clang::driver::JobList const&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*>>&, bool) const (/testing/llvm/bin_main/bin/clang-16+0x2cf80ad)
#42 0x000056486023b3cc clang::driver::Driver::ExecuteCompilation(clang::driver::Compilation&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*>>&) (/testing/llvm/bin_main/bin/clang-16+0x2d073cc)
#43 0x000056485e30fbe5 clang_main(int, char**) (/testing/llvm/bin_main/bin/clang-16+0xddbbe5)
#44 0x00007fc0be031d90 (/lib/x86_64-linux-gnu/libc.so.6+0x29d90)
#45 0x00007fc0be031e40 __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x29e40)
#46 0x000056485e30a465 _start (/testing/llvm/bin_main/bin/clang-16+0xdd6465)
clang-16: error: clang frontend command failed with exit code 134 (use -v to see invocation)
clang version 16.0.0 (https://github.com/llvm/llvm-project.git b50392342be27efdc01d877ba0c357485dc7e918)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /testing/llvm/bin_main/bin
```

**Comment by RKSimon** [comment]
Not reproducible in trunk from the cpp: https://godbolt.org/z/sKTorzrMj

But the IR still crashes in SLP: https://www.godbolt.org/z/PoxPYGx8o

CC @alexey-bataev

**Comment by alexey-bataev** [comment]
Fixed in b1abc2beaff38294e8f4a3d21999b3e8e4c1e6e2

**Comment by dbabokin** [comment]
Thanks for fixing!
